test/unittest/rtps/history/HistoryTests.cpp

```diff
@@ -0,0 +1,255 @@
+            if (it == m_changes.end())
+            {
+                return false;
+            }
+
+            m_changes.erase(it, m_changes.end());
+            return true;
+        }
+
+    private:
+        eprosima::fastrtps::RecursiveTimedMutex mutex_;
+    };
+
+    HistoryTests()
+        : pool(nullptr)
+        , pool_size(10)
+        , max_pool_size(0)
+        , payload_size(128)
+        , memory_policy(MemoryManagementPolicy_t::PREALLOCATED_MEMORY_MODE)
+    {
+    }
+
+    virtual ~HistoryTests()
+    {
+    }
+
+    virtual void SetUp()
+    {
+        pool_size     = get<0>(GetParam());
+        max_pool_size = get<1>(GetParam());
+        payload_size  = get<2>(GetParam());
+        memory_policy = get<3>(GetParam());
+
+        HistoryAttributes att;
+        att.initialReservedCaches = pool_size;
+        att.maximumReservedCaches = max_pool_size;
+        att.payloadMaxSize = payload_size;
+        att.memoryPolicy = memory_policy;
+
+        pool = new HistoryForTest(att);
+    }
+
+    virtual void TearDown()
+    {
+        delete pool;
+    }
+
+    HistoryForTest* pool;
+
+    int32_t pool_size;
+    int32_t max_pool_size;
+    uint32_t payload_size;
+    MemoryManagementPolicy_t memory_policy;
+};
+
+TEST_P(HistoryTests, reserve_cache)
+{
+    CacheChange_t* ch = nullptr;
+
+    uint32_t payload = payload_size;
+    uint32_t size = static_cast<uint32_t>(pool_size);
+    uint32_t max_size = static_cast<uint32_t>(max_pool_size);
+
+    uint32_t num_inserts;
+    if (max_size == 0)
+    {
+        num_inserts = 1000U;
+    }
+    else if (max_size <= size)
+    {
+        max_size = size;
+        num_inserts = size + 1;
+    }
+    else // max_size > size
+    {
+        num_inserts = max_size + 1;
+    }
+
+    for (uint32_t i = 0; i < num_inserts; i++)
+    {
+        uint32_t data_size = i * 16;
+        ASSERT_TRUE(pool->reserve_Cache(&ch, [data_size]() -> uint32_t
+                {
+                    return data_size;
+                }));
+
+        switch (memory_policy)
+        {
+            case MemoryManagementPolicy_t::PREALLOCATED_MEMORY_MODE:
+                ASSERT_EQ(ch->serializedPayload.max_size, payload);
+                break;
+            case MemoryManagementPolicy_t::PREALLOCATED_WITH_REALLOC_MEMORY_MODE:
+                ASSERT_EQ(ch->serializedPayload.max_size, max(payload, data_size));
+                break;
+            case MemoryManagementPolicy_t::DYNAMIC_RESERVE_MEMORY_MODE:
+                ASSERT_EQ(ch->serializedPayload.max_size, data_size);
+                break;
+            case MemoryManagementPolicy_t::DYNAMIC_REUSABLE_MEMORY_MODE:
+                ASSERT_EQ(ch->serializedPayload.max_size, data_size);
+                break;
+        }
+    }
+
+    if (max_size == 0)
+    {
+        ASSERT_TRUE(pool->reserve_Cache(&ch, payload));
+    }
+    else
+    {
+        ASSERT_FALSE(pool->reserve_Cache(&ch, payload));
+    }
+}
+
+TEST_P(HistoryTests, chage_change)
```

Comment:
```suggestion
TEST_P(HistoryTests, change_change)
```

It is also wrong on the original CacheChangePoolTests.cpp, but I cannot comment there because that section is not modified. Could we correct both?

Reply:
Renamed both to `change_reset_on_release`